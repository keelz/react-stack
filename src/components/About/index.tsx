import * as React from 'react';

const About: React.SFC = () => <div className="About">About</div>;

export default About;
