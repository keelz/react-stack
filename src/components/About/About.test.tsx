import * as React from 'react';
import * as Adapter from 'enzyme-adapter-react-16';
import { configure, shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import About from './index';

const adapter = new Adapter();
configure({ adapter });

describe('AppHeader suite', () => {
  it('should match snapshot', () => {
    const tree = shallow(<About />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should be selectable by class .About', () => {
    expect(shallow(<About />).is('.About')).toBe(true);
  });

  it('should mount in a full DOM', () => {
    expect(mount(<About />).find('.About').length).toBe(1);
  });
});
