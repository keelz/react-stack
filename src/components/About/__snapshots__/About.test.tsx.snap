// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader suite should match snapshot 1`] = `
<div
  className="About"
>
  About
</div>
`;
